Magento2 code refactor Changes

nileshdbabar2023 opened the pull request:
**Removing Usage of Object manager**
- Removed usage of object manager from the direct call.
- used respective object creation in the constructor.
- Updated Api call to use directory list.
- constant log added in the API.
- validation check added for the MSI modules present or not.
- Di.xml updated for the file system call and directory list call.
- stock resolver updated.
- stock provider updated.

**Resolved and Fixed Test cases**
- Added fix for the errors of test cases.
- updated test cases with the current implementation.

**File Handling using Magento file system**
- Used methods isExists,fileGetContents instead of direct call of file_get_contents ,file_exists
- Added logging to file.

**Handling and displaying error log**
- Added logging of error for the stock provider
- Exception handling Error log added.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1209442416533376

tejasklevu commented on Helper/LogInfo.php:
```diff
@@ -11,24 +11,51 @@
-        $directory = $objectManager->get('\Magento\Framework\Filesystem\DirectoryList');
-        $logPath = $directory->getPath('log');
+        $logPath = $this->directoryList->getPath(DirectoryList::LOG);
         $logFile = $logPath . '/searchspring_feed.log';
```
Can we put searchspring_feed,klog and other details as constant?

nileshdbabar2023 replied:
Changes Updated to use constant

tejasklevu commented on Helper/LogInfo.php:
```diff
@@ -11,24 +11,51 @@
+        if ($this->fileDriver->isExists($logFile)) {
+            $this->logger->info("File searchspring feed log will be removed from the path:" . $logPath);
             unlink($logFile);
+            $this->logger->info("File removed successfully" . $logPath . '/searchspring_feed.log');
```
Same as above

nileshdbabar2023 replied:
Added changes to use constant

tejasklevu commented on Helper/VersionInfo.php:
```diff
@@ -21,24 +23,41 @@ class VersionInfo extends AbstractHelper
 
         $version = null;
-        $module = $objectManager->get('\Magento\Framework\Module\ModuleListInterface')->getOne(self::MODULE_NAME);
+        $module = $this->moduleListInterface->getOne(self::MODULE_NAME);
```
Possible to get it from composer.json only as getOne also executes getAll() internally.
Example on how we did it in 3.x
https://github.com/klevu/productsearch/blob/3.x/Helper/VersionReader.php

nileshdbabar2023 replied:
Used the composer method mentioned

tejasklevu commented on Model/Feed/DataProvider/Stock/MsiStockProvider.php:
```diff
@@ -145,6 +163,10 @@ public function getStock(array $productIds, int $storeId): array
+                    "Error processing stock data for SKU: {$sku}",
+                    ['exception' => $exception]
+                );
                 continue;
```
Do we need this?

nileshdbabar2023 replied:
Yes we require this .

tejasklevu commented on Model/Feed/DataProvider/Stock/MsiStockProvider.php:
```diff
@@ -166,6 +188,7 @@ public function getStock(array $productIds, int $storeId): array
+
         // we cannot use constructor because MSI module codebase can be removed via composer
         if (is_null($this->getReservationsQuantity)) {
             $this->getReservationsQuantity = ObjectManager::getInstance()
```
The ObjectManager is again used directly, can this be fixed?

tejasklevu reviewed (commented):
I added some comments on specific files which are part of this merge request. Some other things i still noticed:
1. ObjectManager are still being used e.g. MsiStockProvider.php file and many more, can we try to get rid of that?
2. Some of the SQL queries are used e.g.  $condition = '(e.created_at >= :from OR e.updated_at >= :from)'; in file Customer.php. According to Magento, its not best practices. Its not priority thing but can be considered for future changes.

nileshdbabar2023 commented:
> I added some comments on specific files which are part of this merge request. Some other things i still noticed:
> 
> 1. ObjectManager are still being used e.g. MsiStockProvider.php file and many more, can we try to get rid of that?
> 2. Some of the SQL queries are used e.g.  $condition = '(e.created_at >= :from OR e.updated_at >= :from)'; in file Customer.php. According to Magento, its not best practices. Its not priority thing but can be considered for future changes.

Have updated those and removed the usage of the Object manager, except the point number 2 that we will check in the future changes.
